### PR TITLE
Adding frag around else code in react Show directive

### LINKED ISSUE #1079: Passing a null value or empty array in when prop of Show component causes build error
### I am interested in helping provide a fix!

Yes

### Which generators are impacted?

- [ ] All
- [ ] Angular
- [ ] HTML
- [ ] Qwik
- [X] React
- [ ] React-Native
- [ ] Solid
- [ ] Stencil
- [ ] Svelte
- [ ] Vue
- [ ] Web components

### Reproduction case

https://mitosis.builder.io/?outputTab=E4UwhgxgLkA%3D&code=JYWwDg9gTgLgBAbzgVwM4FMDKNroDRyYAWEA7gQGLRwC%2BcAZlBCHAEQACARssADYAm6KADpgEAPQhgOVMFSsA3ACgl6AB6RYcQfQCGyXvHrIAdgGMYYk3ACyATwDCzSCfQmYACjBMwqAJSISnBwZhAmqPARujDocAC8KBjYuB4IQcFwJsggnEKocABccADaALp46cEAbmTovPlFxQDkuk0ETehtcE3AXU0QfchNpek0firBUOgwyFDWHpVwADz8wFUAfIvBS8RkcKREbnEIUTHCWTl5NFsZdRjHNxkZS1RQcOi6ZkTxiKfowjVSHcaJsnmCwQgPIC6gE4utlkQAIzraG8QqIVE0JbiJHra7g8HY16ggn4sEkwmvd6fIjHP7nbK5KCoEGPDKQi5M2Hwpa4zlCdEIflQLE45FkwniYk3bG7UgU5biVYbdJ%2BZQ0IA%3D%3D

### Expected Behaviour

It should not throw build error and should output code for react aswell (It works for qwik/vue).

### Actual Behaviour

It throws build error for react 

### Additional Information

Screenshot 2023-03-28 at 6 59 00 PM

### PR DESCRIPTION
**Pull request by akshatgarg12**
## Description

Add a short description of:
Fixes #1079 
- Added frag around else block of react show directive, updated e2e-app tests with show-for components, updated playwright test
- This fixes the issue where passing For in else was throwing build errors

### CODE REVIEW

File: e2e/e2e-app/tests/example.spec.ts [hunk 1]
```diff
@@ -27,4 +27,11 @@ test.describe('e2e', () => {
 
     // await expect(page.locator('li')).toHaveCount(3);
   });
+  test('show-for component test', async ({ page }) => {
+    await page.goto('/show-for/');
+
+    await page.locator('text=number : 1')
+    await page.locator('text=number : 2')
+    await page.locator('text=number : 3')
```
Comment (samijaber) [inline e2e/e2e-app/tests/example.spec.ts:35]: That's not quiet right. You need to do `await expect(page.locator('text=***')).toBeVisible()`;

File: packages/core/src/generators/react/blocks.ts [hunk 2]
```diff
@@ -106,12 +106,13 @@ const NODE_MAPPERS: {
   },
   Show(json, options, component) {
     const wrap = wrapInFragment(json);
+    const wrapElse = true; // TODO: make this configurable
```
Comment (samijaber) [inline packages/core/src/generators/react/blocks.ts:109]: I think you can implement this by doing:

```suggestion
    const wrapElse = wrapInFragment(json.meta.else as any)
```

Comment (akshatgarg12) [reply packages/core/src/generators/react/blocks.ts:109]: This doesn't work for me

**Comment by akshatgarg12** [comment]
Tests are failing due to `<h1>number : {number}</h1>`  being output as `<h1>number :{number}</h1>` in react. (Notice the space after colon)

**Review by samijaber — APPROVED** [review]

**Comment by samijaber** [comment]
@akshatgarg12 I think you still haven't updated your snapshots either by doing `yarn snapupdate` in `packages/core`

**Review by samijaber — APPROVED** [review]
🙏🏽 🏆

**Pull request merged**